tmt/schemas/plans.yaml

```diff
@@ -0,0 +1,99 @@
+---
+
+#
+# JSON Schema definition for tmt `Plans`
+#
+# https://tmt.readthedocs.io/en/stable/spec/stories.html
+#
+
+$id: /schemas/plans
+$schema: https://json-schema.org/draft-07/schema
+
+type: object
+additionalProperties: false
+
+properties:
+
+  # https://tmt.readthedocs.io/en/stable/spec/core.html#adjust
+  adjust:
+    $ref: "/schemas/core#/definitions/adjust"
+
+  # https://tmt.readthedocs.io/en/stable/spec/plans.html#context
+  # TODO
+  context:
+    $ref: "/schemas/plans#/definitions/context"
+
+  # https://tmt.readthedocs.io/en/stable/spec/core.html#description
+  description:
+    $ref: "/schemas/core#/definitions/description"
+
+  # https://tmt.readthedocs.io/en/stable/spec/plans.html#spec-plans-discover
+  discover:
+    oneOf:
+      - $ref: "/schemas/plans#/definitions/discover_fmf"
+      - $ref: "/schemas/plans#/definitions/discover_shell"
```

Comment:
I would probably go with standalone files for these various "how" objects. There are some shared properties, these schemas can import them from `/schemas/common`, but it'd seem easier to follow the trail with `/schemas/discover/fmf`, `/schemas/discover/shell`, `/schemas/provision/virtual` and so on.

Unlike tests or stories, plans are open to plugins. There can be - and already are - private, out-of-tree plugins, and these should provide their schemas too. With a monolithic `/schemas/plans`, it would be hard to inject such a schema for a custom provision plugin. I imagine that these plugins would "install" their schemas by dropping them into `${tmt_schemas_directory}/provision/` directory, for example - tmt can then do a little "ls" and convert found provision/discover/execute/etc. schemas into a dynamic list to be 1. injected into `/schemas/plans`'s `provision`/`discover`/`execute`/etc. sections, and 2. load them into schema store.

The list of plugins can be dynamic, I can't see this working with a hardcoded list of allowed schemas, that would break at the first encounter with an out-of-tree plugin.

Reply:
yeah, will try this approach, makes sense, thanks for the comments